server/api/crud/model_monitoring/deployment.py

```diff
@@ -288,54 +283,51 @@ def apply_and_create_stream_trigger(
 
         # Get the stream path from the configuration
-        stream_paths = server.api.crud.model_monitoring.get_stream_path(
+        stream_path = server.api.crud.model_monitoring.get_stream_path(
             project=self.project, function_name=function_name
         )
         # set all MM app and infra to have only 1 replica
         function.spec.max_replicas = 1
-        for i, stream_path in enumerate(stream_paths):
-            if stream_path.startswith("kafka://"):
-                topic, brokers = mlrun.datastore.utils.parse_kafka_url(url=stream_path)
-                # Generate Kafka stream source
-                stream_source = mlrun.datastore.sources.KafkaSource(
-                    brokers=brokers,
-                    topics=[topic],
-                )
-                stream_source.create_topics(num_partitions=1, replication_factor=1)
-                function = stream_source.add_nuclio_trigger(function)
+        if stream_path.startswith("kafka://"):
+            topic, brokers = mlrun.datastore.utils.parse_kafka_url(url=stream_path)
+            # Generate Kafka stream source
+            stream_source = mlrun.datastore.sources.KafkaSource(
+                brokers=brokers,
+                topics=[topic],
+            )
+            stream_source.create_topics(num_partitions=1, replication_factor=1)
+            function = stream_source.add_nuclio_trigger(function)
 
-            if not mlrun.mlconf.is_ce_mode():
-                if stream_path.startswith("v3io://"):
-                    if "projects" in stream_path:
-                        stream_args = (
-                            config.model_endpoint_monitoring.application_stream_args
-                        )
-                        access_key = self.model_monitoring_access_key
-                        kwargs = {"access_key": self.model_monitoring_access_key}
-                    else:
-                        stream_args = (
-                            config.model_endpoint_monitoring.serving_stream_args
-                        )
-                        access_key = os.getenv("V3IO_ACCESS_KEY")
-                        kwargs = {}
-                    if mlrun.mlconf.is_explicit_ack_enabled():
-                        kwargs["explicit_ack_mode"] = "explicitOnly"
-                        kwargs["worker_allocation_mode"] = "static"
-                    server.api.api.endpoints.nuclio.create_model_monitoring_stream(
-                        project=self.project,
-                        stream_path=stream_path,
-                        access_key=access_key,
-                        stream_args=stream_args,
-                    )
-                    # Generate V3IO stream trigger
-                    function.add_v3io_stream_trigger(
-                        stream_path=stream_path,
-                        name=f"monitoring_{function_name}_trigger{f'_{i}' if i != 0 else ''}",
-                        **kwargs,
+        if not mlrun.mlconf.is_ce_mode():
```

Comment:
I know it's not your relevant to your change, but still I would like to comment that it's better to fail here if it's ce_mode. Current code has 2 cons:
1. fails silently on an unsupported case
2. adds one more indentation which is less readable